doc/SONiC_Port_Mirroring_HLD.md

```diff
@@ -0,0 +1,319 @@
+    - CLI to allow mirror session configuration only with destination port.
+
+
+## 1.2 Configuration and Management Requirements
+- Existing CLI 'config mirror_session add/remove'to be extended to include source port/portchannel.
+- Existing CLI 'config mirror_session add/remove' to be extended to include destination port/portchannel.
+- Existing CLI 'show mirror session' is extended to support all flavors of mirror sessions.
+
+
+## 1.3 Scalability Requirements
+- Up to max ASIC capable mirror sessions to be supported.
+- Once max mirror sessions are created and user attempts to create new session, error will be logged in syslog.
+
+
+## 1.4 Warm Boot Requirements
+- Mirroring functionality should continue to work across warm reboot.
+
+To support planned system warm boot.
+To support SWSS docker warm boot.
+
+
+# 2 Functionality
+
+Refer section 1
+
+## 2.2 Functional Description
+Refer section 1.1
+
+## 2.3 Functional Description
+
+Mirroring to destination VLAN (RSPAN) is not supported in this release.
+
+# 3 Design
+## 3.1 Overview
+
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+
+Existing table PORT_MIRROR_TABLE is enhanced to accept new source and destination configuration options  in the configuration database. This table is filled by the management framework.
+
+#### CONFIG_PORT_MIRROR_TABLE
+
+    ;Configure SPAN/ERSPAN mirror session.
+    ;storm control type - broadcast / unknown-unicast / unknown-multicast
+    key       = PORT_MIRROR_TABLE:mirror_session_name ; mirror_session_name is
+                                                      ; unique session
+                                                      ; identifier
+    ;field  = value
+    destination_port = PORT_TABLE:ifname    ; ifname must be unique across PORT TABLE.
+    source_port = PORT_TABLE:ifname    ; ifname must be unique across PORT,INTF,LAG TABLES
+    direction     = ingress or egress or both           ; Direction ingress or egress or both.
+
+    mirror_session_name = 1*255VCHAR
+
+### 3.2.2 APP_DB
+No tables are introduced in APP_DB
+### 3.2.3 STATE_DB
+No tables are introduced in STATE_DB.·
+
+### 3.2.4 ASIC_DB
+No changes are introduced in ASIC_DB.·
+### 3.2.5 COUNTER_DB
+No changes are introduced in COUNTER_DB.·
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+
+Mirror Orchestration agent is modified to support this feature:
+   - Handle both SPAN and ERSPAN sessions separately·
+   - No changes to ERSPAN functionality.
+   - Configure mirror session based on CONFIG_DB parameters.
+   - Port mirror session is activated 
+   - Populates the mirror attribute SAI structures and pushes the entry to ASIC_DB.·
+
+## 3.4 SAI
+Mirror SAI interface APIs are already defined. More details about SAI API and attributes are described below SAI Spec @
+
+https://github.com/opencomputeproject/SAI/blob/master/inc/saimirror.h
+
+## 3.5 CLI
+### 3.5.1 Data Models
+Custom Yang model will be introduced for this feature.
```

Comment:
can you add description for the yang model?

Reply:
Sure. I will add and update the doc

Reply:
Updated doc with both SONiC yang and openconfig extension model also